presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java

```diff
@@ -283,6 +284,11 @@ public SystemSessionProperties(
                         featuresConfig.isLegacyOrderBy(),
                         false),
                 booleanSessionProperty(
+                        SUPPORT_NULL_ON_LHS_OF_SEMI_JOIN,
```

Comment:
It's not obvious for me that it is almost useless in practice. You can easily come up with query where predicate pushdown will make a great difference for semi join, e.g:
```
SELECT x IN (SELECT y FROM very_large_table) FROM ... WHERE x=3
```

Why it would be almost useless in practice for semi join and useful for inner join for example?